gdal/ogr/ogrutils.cpp

```diff
@@ -214,6 +338,37 @@ void OGRFormatDouble( char *pszBuffer, int nBufferLen, double dfVal,
     }
 }
 
+
+/// Simplified OGRFormatDouble that can be made to adhere to provided
+/// options.
+std::string OGRFormatDouble(double val, OGRWktOptions opts)
+{
+    // So to have identical cross platform representation.
+    if( CPLIsInf(val) )
+        return (val > 0) ? "inf" : "-inf";
+    if( CPLIsNan(val) )
+        return "nan";
+
+    std::ostringstream oss;
+    if (opts.format == OGRWktFormat::F ||
+        (opts.format == OGRWktFormat::Default && fabs(val) < 1))
+        oss << std::fixed;
+    else
+    {
+        // Uppercase because OGC spec says capital 'E'.
+        oss << std::defaultfloat << std::uppercase;
+        opts.round = false;
+    }
+    oss << std::setprecision(opts.precision);
```

Comment:
`oss.imbue(std::locale::classic());` missing

Reply:
What does that do?

Reply:
Force the C locale so that decimal point is always output instead of comma as in some locales (like fr_FR.UTF-8)